ddtrace/contrib/vertica/patch.py

```diff
@@ -0,0 +1,269 @@
+                        "operation_name": "vertica.fetchall",
+                        "span_type": "vertica",
+                        "span_end": fetch_span_end,
+                    },
+                    "nextset": {
+                        "operation_name": "vertica.nextset",
+                        "span_type": "vertica",
+                        "span_end": fetch_span_end,
+                    },
+                }
+            },
+        },
+    },
+)
+
+
+def patch():
+    global _PATCHED
+    if _PATCHED:
+        return
+
+    _install(config.vertica)
+    _PATCHED = True
+
+
+def unpatch():
+    global _PATCHED
+    if _PATCHED:
+        _uninstall(config.vertica)
+        _PATCHED = False
+
+
+def _uninstall(config):
+    for patch_class_path in config["patch"]:
+        patch_mod, _, patch_class = patch_class_path.rpartition(".")
+        mod = importlib.import_module(patch_mod)
+        cls = getattr(mod, patch_class, None)
+
+        if not cls:
+            log.debug(
+                """
+                Unable to find corresponding class for tracing configuration.
+                This version may not be supported.
+                """
+            )
+
+        for patch_routine in config["patch"][patch_class_path]["routines"]:
+            unwrap(cls, patch_routine)
+
+
+def _find_routine_config(config, instance, routine_name):
+    """Attempts to find the config for a routine based on the bases of the
+    class of the instance.
+    """
+    bases = instance.__class__.__mro__
+    for base in bases:
+        full_name = "{}.{}".format(base.__module__, base.__name__)
+        if full_name not in config["patch"]:
+            continue
+
+        config_routines = config["patch"][full_name]["routines"]
+
+        if routine_name in config_routines:
+            return config_routines[routine_name]
+    return {}
+
+
+def _install_init(patch_item, patch_class, patch_mod, config):
+    patch_class_routine = "{}.{}".format(patch_class, "__init__")
+
+    # patch the __init__ of the class with a Pin instance containing the defaults
+    @wrapt.patch_function_wrapper(patch_mod, patch_class_routine)
+    def init_wrapper(wrapped, instance, args, kwargs):
+        wrapped(*args, **kwargs)
+
+        # create and attach a pin with the defaults
+        Pin(
+            service=config["service_name"],
+            app=config["app"],
+            app_type=config["app_type"],
+            tags=config.get("tags", {}),
+            tracer=config.get("tracer", ddtrace.tracer),
+            _config=config["patch"][patch_item],
+        ).onto(instance)
+
+
+def _install_routine(patch_routine, patch_class, patch_mod, config):
+    patch_class_routine = "{}.{}".format(patch_class, patch_routine)
+
+    @wrapt.patch_function_wrapper(patch_mod, patch_class_routine)
+    def wrapper(wrapped, instance, args, kwargs):
+        # TODO?: remove Pin dependence
+        pin = Pin.get_from(instance)
+
+        # TODO: inheritance problem with pins:
```

Comment:
You can likely remove this TODO, but keep some form of this comment so we know the problem/what we are doing.